### ISSUE TITLE
Mesh-to-octree conversion consumes a lot of memory

### ISSUE THREAD

**Issue by joeylitalien** [NVIDIAGameWorks/kaolin-wisp#9]
The `mesh_to_octree()` function in `ops/spc/conversions.py` has a number of surface samples hardcoded to 100M for voxel initialization. This consumes a large amount of memory, so much that even on a half-decent GPU like a Titan V (12GB), the vanilla NGLOD-SDF config will fail on a simple Armadillo.

One way to address this would be to expose this number directly in a config file (at the risk of possibly undersampling the shape). Another option would be to catch the runtime error to reduce the load to something more reasonable (e.g., 50M), which could be handled by a simple split-concat:

```python
try:
   samples = mesh_ops.sample_surface(vertices.cuda(), faces.cuda(), 100000000)[0]
except RuntimeError as e:
   samples = [mesh_ops.sample_surface(vertices.cuda(), faces.cuda(), 10000000)[0] for _ in range(5)]
   samples = torch.cat(samples, dim=0)
   # Mention something about reducing number of samples to fit memory
```

Anyway, just a tip to make this application a bit more accessible to newcomers :)

**Comment by 3a1b2c3** [comment]
+1 Tried to use it with a sphere and its a 1 gb

**Comment by orperel** [comment]
Thanks for the feedback @joeylitalien !
This is addressed in #26 
You use `--num-samples-on-mesh` to control the samples count
